src/algorithms/dqns/dqn.jl

```diff
@@ -102,18 +106,28 @@ function RLBase.update!(learner::DQNLearner, batch::NamedTuple)
     loss_func = learner.loss_func
     n = learner.sampler.n
     batch_size = learner.sampler.batch_size
+    double = learner.double
     D = device(Q)
 
     s, a, r, t, s′ = (send_to_device(D, batch[x]) for x in SARTS)
     a = CartesianIndex.(a, 1:batch_size)
 
-    target_q = Qₜ(s′)
+    if double
+        action_values = Q(s′)
+    else
+        action_values = Qₜ(s′)
+    end
+
     if haskey(batch, :next_legal_actions_mask)
         l′ = send_to_device(D, batch[:next_legal_actions_mask])
-        target_q .+= ifelse.(l′, 0.0f0, typemin(Float32))
+        action_values .+= ifelse.(l′, 0.0f0, typemin(Float32))
     end
 
-    q′ = dropdims(maximum(target_q; dims = 1), dims = 1)
+    selected_actions = [(findmax(action_values[:,i]; dims=1))[2][1] for i in 1:batch_size]
+    target_q = Qₜ(s′)
+    q′ = [target_q[:,i][selected_actions[i]][1] for i in 1:batch_size]
```

Comment:
Could you test it again with GPU enabled? You may need vectorized operation instead of list comprehension here.

Reply:
Also, make sure the code logic is the same with the original one when `double` is set to `false`